src/scripting/ScriptContext.cpp

```diff
@@ -145,8 +145,18 @@ ScriptContext::ScriptContext(LuaSandbox& aLuaSandbox, const std::filesystem::pat
                 inputTypeStr, acID, acDisplayName, bindIt->DisplayName);
             return;
         }
+        sol::function vmHandler = aCallback;
+        if (acIsHotkey)
+        {
+            vmHandler = MakeSolFunction(lockedState, [aCallback](bool isPressed) {
+                if (isPressed)
+                {
+                    aCallback();
+                }
+            });
+        }
```

Comment:
Here is the code, that handles hotkeys. Personaly, I would rather expose single interface for both hotkey and input callback - `void callback(bool isPressed)`. But if you see an issue with that, I will make two separate fields `isHotkey` and `callback`. What do you prefer?

Reply:
Hotkeys react on key up as mentioned before, otherwise they would trigger in wrong moment (imagine binds like ctrl + shift and ctrl + shift + a, it would register both if it was on key down or completely ignore the longer one)

Reply:
Yes, that is how CET handles hotkeys internally.
Let's imagine communication between two mods: 
In "clientMod":
```registerHotkey("actionId", "Custom Action", function() 
...
end)```

In "InputMod" when certain game input action event is detected:
```GetMods()["clientMod"].inputHandlers["actionId"].callback(true) -- pressed
```

In "InputMod" when certain game input action finished:
```GetMods()["clientMod"].inputHandlers["actionId"].callback(false) -- released
```
I want the "InputMod" to not care about what callback it is, hotkey or not.
So I am not changing what CET is doing with hotkeys, just adding new api for mods like "InputMod".
Alternative would be to do it like this:

```
local handler = GetMods()["clientMod"].inputHandlers["actionId"]
if handler.isHotkey then
    handler.callback()
else
   handler.callback(true)
end
```

Reply:
Yeah like that seems fine. Only issue I had with having it in same handler is that hotkey should trigger when key is up (so when false is passed into the function, you react when key is down so when true is passed in)

Think it would just confuse things tbh, when real behavior is one but this is different.

Reply:
Yes, your point makes sense to me. You are thinking about how it already works in CET. So it would be confusing if hotkeys would trigger on press if they are already triggering on release.

But I am thinking about what interface we are giving to the mods. So my logic is:
 - when you want activation: call `callback(true)`
 - when you want deactivation (if that is even supported): call `callback(false)`
So in that sense, hotkeys do not support deactivation. That would be the only difference from inputs.

After your proposal 
> (so when false is passed into the function, you react when key is down so when true is passed in)
Activation would look like:
```
handler.callback(not handler.isHotkey)
```
If you insist on changing that, I would rather remove the if statement, and leave just 
```
sol::function vmHandler = aCallback;
```

Reply:
Well, I would because I still think that this should basically emulate the lower level of things if it should be exposed. Lower level behaves like it behaves. Trying to flip the behaviour wouldn't lead to any good in the long run imho. 

I suppose as you now expose the isHotkey, just use the calback directly. Anyone can resolve it on the Lua side how they want afterwards.